packages/website/src/features/Packages/Tabs/InteractTab.tsx

```diff
@@ -137,6 +137,30 @@ export const InteractTab: FC<{
       highlightedData = allContracts;
     }
 
+    const uniqueAddresses = new Set();
+    for (const contractData of highlightedData) {
+      uniqueAddresses.add(contractData.contractAddress);
+    }
+
+    for (const uniqueAddress of uniqueAddresses) {
+      const excessContracts = highlightedData.filter(
```

Comment:
this is really inefficient. why not just check to see if the unique address is part of the set at the beginning, and then `continue` if it matches an existing address already?

also to improve readability, consider using lodash to convert this code to 1 succinct line (not everyone agrees :( tho)